### PR TITLE
FILESYSTEM-TESTS: Consolidate xfstests test cases

### PR DESCRIPTION
**Pull request by adriansuhov**
xfstesting.sh (previously perf_xfstesting.sh) was refactored. Changes
made:
- Renamed perf_xfstesting.sh to xfstesting.sh. This isn't a perf test so
we dropped the 'perf_' prefix
- Replaced 'CIFS' testing with 'Generic' filesystem tests. It's more
valuable from a testing perspective since it covers a lot more tests and
use cases.
- NVMe support - 4 new NVMe functional tests were added:
NVME-FILE-SYSTEM-VERIFICATION-GENERIC
NVME-FILE-SYSTEM-VERIFICATION-XFS
NVME-FILE-SYSTEM-VERIFICATION-EXT4
NVME-FILE-SYSTEM-VERIFICATION-BTRFS
- Removed xfstests-config.config. The config file is constructed in PS script.
- Consolidated bash script:
---- It now supports Debian/Ubuntu/RHEL/CentOS distros
---- Removed duplicate code
---- Added excluded tests: without this list, xfstests run would have
hanged.
- Consolidated Powershell script:
---- Added a timeout of 3 hours. If the bash script exceeds this
timeout, the test will be aborted.
---- Better log collect - Collect-TestLogs function is used to download
the logs from the test VM

# Test results
[LISAv2 Test Results Summary]
Test Run On           : 02/03/2019 08:24:53
ARM Image Under Test  : Redhat : RHEL : 7.5 : latest
Total Test Cases      : 4 (4 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:3:12

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 FILE-SYSTEM-VERIFICATION-TESTS-GENERIC                                            PASS                 96.1
    2 FILE-SYSTEM-VERIFICATION-TESTS-XFS                                                PASS                66.74
    3 FILE-SYSTEM-VERIFICATION-TESTS-EXT4                                               PASS                 5.41
    4 FILE-SYSTEM-VERIFICATION-TESTS-BTRFS                                              PASS                17.76

[LISAv2 Test Results Summary]
Test Run On           : 02/02/2019 14:19:18
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Total Test Cases      : 4 (4 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:1:50

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 NVME-XFSTESTS-GENERIC                                                             PASS                46.63
    2 NVME-XFSTESTS-XFS                                                                 PASS                29.17
    3 NVME-XFSTESTS-EXT4                                                                PASS                 8.78
    4 NVME-XFSTESTS-BTRFS                                                               PASS                13.25

* NVME test case names were changed meanwhile - the new names are the same as stated in this PR description above

### CODE REVIEW

File: Testscripts/Linux/xfstesting.sh [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    # Put remaining params in the config file
+    echo "SCRATCH_DEV=/dev/${secondary_partition}" >> ${XFSTestConfigFile}
+    echo "SCRATCH_MNT=/root/${secondary_mountpoint}" >> ${XFSTestConfigFile}
+}
+
+Main() {
+    if [ -e ${XFSTestConfigFile} ]; then
+        LogMsg "${XFSTestConfigFile} file is present."
+    else
+        LogErr "Error: missing ${XFSTestConfigFile} file"
```
Comment (chvalean) [inline Testscripts/Linux/xfstesting.sh]: remove error:

File: Testscripts/Windows/FILE-SYSTEM-VERIFICATION-TESTS.ps1 [hunk 2]
```diff
@@ -1,15 +1,33 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the Apache License.
-param([object] $AllVmData, [object] $CurrentTestData)
+#
+# Description:
+#     This Powershell script will run xfstexting.sh bash script
+#     - It will construct the config file needed by xfstesting.sh
+#     - It will start xfstesting.sh. Max allowd run time is 3 hours.
```
Comment (chvalean) [inline Testscripts/Windows/FILE-SYSTEM-VERIFICATION-TESTS.ps1]: allowed*

File: Testscripts/Linux/xfstesting.sh [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+ConfigureXFSTestTools() {
+    GetDistro
+    case "$DISTRO" in
+        ubuntu*|debian*)
+            until dpkg --force-all --configure -a; sleep 10; do echo 'Trying again...'; done
+            pack_list="libacl1-dev libaio-dev libattr1-dev libgdbm-dev libtool-bin libuuid1 libuuidm-ocaml-dev sqlite3 uuid-dev uuid-runtime xfslibs-dev zlib1g-dev"
+        ;;
+
+        redhat*|centos*)
```
Comment (chvalean) [inline Testscripts/Linux/xfstesting.sh]: also add fedora here

File: Testscripts/Windows/FILE-SYSTEM-VERIFICATION-TESTS.ps1 [hunk 4]
```diff
@@ -1,15 +1,33 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the Apache License.
-param([object] $AllVmData, [object] $CurrentTestData)
+#
+# Description:
+#     This Powershell script will run xfstexting.sh bash script
```
Comment (chvalean) [inline Testscripts/Windows/FILE-SYSTEM-VERIFICATION-TESTS.ps1]: testing*

File: Testscripts/Windows/FILE-SYSTEM-VERIFICATION-TESTS.ps1 [hunk 5]
```diff
@@ -22,37 +40,59 @@ function Main {
...
+            }
+            Write-LogInfo "xfstesting.sh is still running!"
         }
-        Copy-RemoteFiles -download -downloadFrom $allVMData.PublicIP -files "XFSTestingConsole.log" -downloadTo $LogDir -port $allVMData.SSHPort -username "root" -password $password
-        $XFSTestingConsole = Get-Content "$LogDir\XFSTestingConsole.log"
 
-        if ($XFSTestingConsole -imatch "Passed all") {
-            $testResult = "PASS"
+        # Get logs. An extra check for the previous $state is needed
+        # The test could actually be hanged. If state.txt is showing
```
Comment (chvalean) [inline Testscripts/Windows/FILE-SYSTEM-VERIFICATION-TESTS.ps1]: could actually hang*

**Comment by adriansuhov** [comment]
**NVMe Ubuntu 16.04 results**
[LISAv2 Test Results Summary]
Test Run On           : 02/04/2019 10:25:03
ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : latest
Total Test Cases      : 4 (4 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:1:50

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 NVME-FILE-SYSTEM-VERIFICATION-GENERIC                                             PASS                57.32
    2 NVME-FILE-SYSTEM-VERIFICATION-XFS                                                 PASS                27.57
    3 NVME-FILE-SYSTEM-VERIFICATION-EXT4                                                PASS                 7.22
    4 NVME-FILE-SYSTEM-VERIFICATION-BTRFS                                               PASS                10.62

**Pull request merged**